pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -1127,6 +1126,13 @@ func (m *kubeGenericRuntimeManager) computeInitContainerActions(pod *v1.Pod, pod
 					changes.InitContainersToStart = append(changes.InitContainersToStart, i+1)
 				}
 
+				// Restart running sidecar containers which have had their definition changed.
+				if _, _, changed := containerChanged(container, status); changed {
+					message = fmt.Sprintf("%s definition changed, will be restarted", message)
+					restartContainer = true
```

Comment:
no reason here?

Reply:
There does not currently exist a reason when a container is restarted due to image/definition change, this is to mimic how it is implemented for regular containers. https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L992

Probably there should be a reason both for regular and sidecars, maybe added here:  https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L439